[CI] (T3K) T3000 e2e tests: t3000-e2e-tests / t3k_fabric_stability_tests – deterministic failure

*Description last updated: **2026-03-23** — refreshed from the three most recent completed `main` runs for this job and their logs.*

**Workflow:** (T3K) T3000 e2e tests (`t3000-e2e-tests.yaml`)

**Job:** `t3000-e2e-tests / t3k_fabric_stability_tests`

This job has failed on **main** in the **latest 3 completed workflow runs** (same terminal failure in all three).

### Failing step / command

CI runs:

`TT_METAL_CLEAR_L1=1 build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_stability.yaml`

### Deterministic failure (from logs)

- **Test case:** `FlowControlLinear_size_1024_num_links_1_ntype_unicast_write_ftype_unicast` (`test_tt_fabric.cpp`)
- **Terminal error:** uncaught `std::runtime_error`: **`Timeout waiting for Ethernet core service remote IO request.`**
- **Process exit:** **134** (SIGABRT after `terminate` from the uncaught exception)
- **Stack (representative):** `FDMeshCommandQueue::finish` / `finish_nolock` in `libtt_metal.so` while waiting after `Launching programs` / `Waiting for programs`

### Most recent failing job

https://github.com/tenstorrent/tt-metal/actions/runs/23424715690/job/68137575286

**Commit:** `a12ecfd86eec8f68c0218b8617d6b028a485fdc7`

### Latest 3 failing job instances (newest first)

- https://github.com/tenstorrent/tt-metal/actions/runs/23424715690/job/68137575286 — `a12ecfd86eec8f68c0218b8617d6b028a485fdc7`
- https://github.com/tenstorrent/tt-metal/actions/runs/23397333194/job/68063078749 — `c26b37ab7555f460fcc8598a8ea381ea9ba673a7`
- https://github.com/tenstorrent/tt-metal/actions/runs/23373779569/job/68002607676 — `5585089575ccd6bd85cd6b4cf2bcd1ecb9a0e87b`

### Error excerpt (log)

```text
Running Test: FlowControlLinear_size_1024_num_links_1_ntype_unicast_write_ftype_unicast
terminate called after throwing an instance of 'std::runtime_error'
  what():  Timeout waiting for Ethernet core service remote IO request.
Signal: Aborted (6)
##[error]Process completed with exit code 134.
```

### Reproduction (T3000 / Wormhole N300 `wh_llmbox`)

From repo root, on hardware matching CI:

1. Checkout a failing commit (e.g. latest above):

```bash
git checkout a12ecfd86eec8f68c0218b8617d6b028a485fdc7
```

2. Build per INSTALLING.md.

3. Run:

```bash
TT_METAL_CLEAR_L1=1 build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_stability.yaml
```

### Temporary mitigation

None linked on this issue yet.

### Historical context (March 2026)

Earlier failures on main for this job (e.g. runs `23039080879`, `22989496522`, `22939832326` around commit `35c50b3327e76772d18500a227ac5a6fda55b705`) showed the same **exit code 134** pattern; current logs identify the **Ethernet core service remote IO timeout** as the concrete terminal cause.

--- a/tests/pipeline_reorg/t3k_e2e_tests.yaml
+++ b/tests/pipeline_reorg/t3k_e2e_tests.yaml
@@ -50,7 +50,8 @@
   team: fabric
 
 - name: t3k_fabric_stability_tests
-  cmd: TT_METAL_CLEAR_L1=1 build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_stability.yaml
+  # Setting TT_METAL_OPERATION_TIMEOUT_SECONDS=0 to disable test timeouts and let the tests run to completion, as some of the stability tests can take a long time to complete.
+  cmd: TT_METAL_OPERATION_TIMEOUT_SECONDS=0 build/test/tt_metal/perf_microbenchmark/routing/test_tt_fabric --test_config tests/tt_metal/tt_metal/perf_microbenchmark/routing/test_fabric_stability.yaml
   skus:
     wh_llmbox:
       timeout: 120

